airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -504,10 +501,25 @@ public KubePodProcess(final boolean isOrchestrator,
         .endSpec()
         .build();
 
+    exitCodeFuture = new CompletableFuture<>();
+
     LOGGER.info("Creating pod...");
     val start = System.currentTimeMillis();
+
     this.podDefinition = fabricClient.pods().inNamespace(namespace).createOrReplace(pod);
 
+    // create the watch before the initialization runs
```

Comment:
nit: can we add a comment explaining why the watcher is useful? we can direct readers to the watcher javadocs for more info.